content/articles/lxc-vs-docker-what-is-the-difference-and-why-docker-is-better/index.md

```diff
@@ -0,0 +1,109 @@
+
+Another issue is that all the guest operating instances are running on a single machine. In case of any boot-up processes, or whenever you want to restart service for some reason, you have to reboot that entire computer. depending on what environment you are you're in, a couple of extra seconds matter
+
+#### LXC virtual environments (VE)
+
+On the other hand, [LXC](https://www.youtube.com/watch?v=CWmkSj_B-wo) (Linux Containers) is an operating system-level virtualization solution that enables the creation and operation of many isolated Linux virtual environments (VE) on a single centralized host. Containers, which are isolation levels, can isolate certain apps or simulate a fully different host.
+
+LXC mitigates VM disadvantages. Linux containers enable the host CPU to effectively divide memory allocations into confinement levels known as namespaces. In comparison, a VM contains the whole operating system and machine setup/emulators, such as the hard disk, virtual CPUs, and network interfaces. Its whole enormous virtualized environment typically takes some time to boot and consumes a large amount of RAM and CPU. LXC virtual environment has no hardware preload emulation. Each virtual environment (an OS or an application) is loaded in a container and executes without any additional overhead and no hardware emulation. This means no penalty from software with limited memory. In the end, LXC will improve the performance of the bare metal as it only bundles the OS/application that is required.
+
+![LXC container](/engineering-education/lxc-vs-docker-what-is-the-difference-and-why-docker-is-better/lxc-container.png)
+
+[Image source](https://www.polarsparc.com/xhtml/LXC.html)
+
+In simple terms, in LXC virtual environments (VE), you create containers for the service the virtual OS, application, or server wants to provide. At the same time, the underlying hardware resources and kernel is shared from all the containers. For example, Let's say you have a container running a LAMP server (Linux, APACHE, MYSQL PHP). In this case, the software installed, the configuration files, and the IP address are all virtualized. However, the container doesn't have the kernel and the RAM (hardware resources) directly associated with it. It is operating on the lower level of the base metal.
+
+In other terms, Let's say you have three different containers, I.e., a LAMP server, a DHCP container, and a DNS container. In this case, all the configuration files are contained within this container. And they all share the underlying resources. In this case, resource allocation will be much easier as compared to VMs. If the LAMP server needs alot of resources, it can be allocated to it. So if the DNS container does need alot of RAM now, and the LAMP server is demanding additional RAM now, that memory can be assigned to the LAMP server to speed up this container's services. Resources are allocated according to the need of every container.
```

Comment:
```suggestion
In other terms, Let's say you have three different containers, I.e., a LAMP server, a DHCP container, and a DNS container. In this case, all the configuration files are contained within this container. And they all share the underlying resources. In this case, resource allocation will be much easier as compared to VMs. If the LAMP server needs a lot of resources, it can be allocated to it. So if the DNS container doesn't need a lot of RAM, and the LAMP server is demanding additional RAM now, that additional memory not used in the DNS server can be assigned to the LAMP server to speed up this container's services. Resources are allocated according to the need of every container.
```